src/rat/ee_utils/ee_aec_file_creator.py

```diff
@@ -24,67 +28,479 @@ def grouper(iterable, n, *, incomplete='fill', fillvalue=None):
+        # Load GRWL data
+        grwl = gpd.read_file(grwl_fp)
+        # Check if GRWL intersects with reservoir geometry
+        intersection = grwl.intersects(reservoir.union_all())
+    else:
+        intersection = np.array([False])
+
+    if intersection.any():
+        method = 'grwl_intersection'
+        print("GRWL intersects with reservoir geometry.")
+        closest_point_to_dam = get_closest_point_to_dam(
+            reservoir, dam_location,
+            buffer_distance=90,  # 90 m buffer from reservoir = 3 pixels
+            grwl_fp=grwl_fp
+        )
+        
+        # Extract latitude and longitude of the closest point
+        lat, lon = closest_point_to_dam.y, closest_point_to_dam.x
+        
+        # Calculate the dam bottom elevation using the dam_bottom_dem_percentile function
+        stats = dam_bottom_dem_percentile(closest_point_to_dam, buffer_distance=250)
+        
+        # Extract the 1 percentile elevation as the dam bottom elevation
+        dam_bottom_elevation = stats['dem_p1']
+    else:
+        method = 'dam_location'
+        if grwl_fp is None:
+            print("GRWL dataset is not provided. You can download global_data which includes GRWL.")
+        else:
+            print("GRWL does not intersect with reservoir geometry.")
+        
+        # Calculate the dam bottom elevation using the dam_bottom_dem_percentile function with a 500 m buffer
+        stats = dam_bottom_dem_percentile(dam_location, buffer_distance=buffer_distance)
+        
+        # Extract the 1 percentile elevation as the dam bottom elevation
+        dam_bottom_elevation = stats['dem_p1']
+
+    return dam_bottom_elevation, method
+
+
+def extrapolate_reservoir(
+    reservoir, dam_location, reservoir_name, dam_height, aec, aev_save_dir, buffer_distance=500,
+    grwl_fp=None
+):
+    """
+    Extrapolates the reservoir's Area-Elevation-Capacity (AEC) curve by fitting a polynomial to observed data and adds a column for storage.
+    - The function first calculates the dam bottom elevation using the specified percentile or GRWL intersection.
+    - It then fits polynomials of degrees 2 and 1 to the observed AEC data, stopping at the first successful fit.
+    - The predicted AEC curve is generated and saved to a CSV file in the specified directory.
+    - The initial guess and polynomial equation are stored as comments in the CSV file.
+    
+    Parameters:
+        reservoir (gpd.GeoSeries): GeoSeries containing the reservoir geometry.
+        dam_location (shapely.geometry.point.Point): The location of the dam as a point geometry.
+        reservoir_name (str): Name of the reservoir.
+        dam_height (float): Height of the dam.
+        aec (pd.DataFrame): DataFrame containing observed AEC data with columns 'CumArea' and 'Elevation'.
+        aev_save_dir (str): Directory path to save the predicted AEC file.
+        buffer_distance (int, optional): Buffer distance around the dam. Default is 500.
```

Comment:
same thing goes for this buffer_distance. We should make it clear that it will be used if there is no intersection of reservoir geometry with GRWL.